diesel_codegen/src/schema_inference/pg.rs

```diff
@@ -38,11 +40,11 @@ table! {
 }
 
 pub fn determine_column_type(cx: &mut ExtCtxt, attr: &ColumnInformation) -> P<ast::Ty> {
-    let tpe = if attr.type_name.starts_with("_") {
-        let subtype = str_to_ident(&capitalize(&attr.type_name[1..]));
+    let tpe = if attr.type_name.ends_with("[]") {
```

Comment:
I don't think this needs to be on the `ColumnInformation` struct. Ultimately we pass the primary key name independently of where that struct is used.

Two joins is high on the list, but it would be incredibly painful to use until some changes land in Rust or I figure out a workaround.